Sources/TuistCore/MetadataProviders/FrameworkMetadataReader.swift

```diff
@@ -0,0 +1,199 @@
+        var uuid: UUID?
+
+        if is64(magic) {
+            var header: mach_header_64 = binary.read()
+            if shouldSwap(magic) {
+                swap_mach_header_64(&header, NX_UnknownByteOrder)
+            }
+            binaryArchitecture = try readBinaryArchitecture(cputype: header.cputype, cpusubtype: header.cpusubtype)
+            binaryLinking = header.filetype == MH_DYLIB ? .dynamic : .static
+            numOfCommands = header.ncmds
+        } else {
+            var header: mach_header = binary.read()
+            if shouldSwap(magic) {
+                swap_mach_header(&header, NX_UnknownByteOrder)
+            }
+            binaryArchitecture = try readBinaryArchitecture(cputype: header.cputype, cpusubtype: header.cpusubtype)
+            binaryLinking = header.filetype == MH_DYLIB ? .dynamic : .static
+            numOfCommands = header.ncmds
+        }
+
+        for _ in 0 ..< numOfCommands {
+            let currentOffset = binary.currentOffset
+            var loadCommand: load_command = binary.read()
+
+            if shouldSwap(magic) {
+                swap_load_command(&loadCommand, NX_UnknownByteOrder)
+            }
+
+            guard loadCommand.cmd == LC_UUID else {
+                binary.seek(to: currentOffset + UInt64(loadCommand.cmdsize))
+                continue
+            }
+
+            binary.seek(to: currentOffset)
+
+            var uuidCommand: uuid_command = binary.read()
+
+            if shouldSwap(magic) {
+                swap_uuid_command(&uuidCommand, NX_UnknownByteOrder)
+            }
+
+            uuid = UUID(uuid: uuidCommand.uuid)
+            break
+        }
+
+        return (binaryArchitecture, binaryLinking, uuid)
+    }
+
+    private func readBinaryArchitecture(cputype: cpu_type_t, cpusubtype: cpu_subtype_t) throws -> BinaryArchitecture {
+        guard
+            let archInfo = NXGetArchInfoFromCpuType(cputype, cpusubtype),
+            let arch = BinaryArchitecture(rawValue: String(cString: archInfo.pointee.name))
+        else {
+            throw PrecompiledMetadataProviderError.architecturesNotFound(binaryPath)
+        }
+        return arch
+    }
+
+    private func readArchiveFormatIfAvailable() {
+        let currentOffset = binary.currentOffset
+        let magic = binary.readData(ofLength: 8)
+        binary.seek(to: currentOffset)
+
+        guard String(data: magic, encoding: .ascii) == archiveFormatMagic else { return }
+
+        binary.seek(to: archiveHeaderSizeOffset)
+        guard let sizeString = binary.readString(ofLength: 10) else { return }
+
+        let size = strtoul(sizeString, nil, 10)
+        binary.seek(to: 8 + sizeOfArchiveHeader + UInt64(size))
+
+        guard let name = binary.readString(ofLength: 16) else { return }
+        binary.seek(to: binary.currentOffset - 16)
+
+        if nameString.hasPrefix(archiveExtendedFormat) {
+            let nameSize = strtoul(String(nameString.dropFirst(3)), nil, 10)
```

Comment:
If I followed along correctly:

```suggestion
            let nameSize = strtoul(String(name.dropFirst(3)), nil, 10)
```

Reply:
heheh that's what last minute refactor does